pkg/prometheus/promcfg.go

```diff
@@ -4092,3 +4066,11 @@ func getLowerByteSize(v *monitoringv1.ByteSize, cpf *monitoringv1.CommonPromethe
 func isByteSizeEmpty(v *monitoringv1.ByteSize) bool {
 	return v == nil || *v == ""
 }
+
+func (cg *ConfigGenerator) addFiltersToYaml(cfg yaml.MapSlice, minimumVersion string, filters []monitoringv1alpha1.Filter) yaml.MapSlice {
+	if filters == nil {
+		return cfg
+	}
+
+	return cg.WithMinimumVersion(minimumVersion).AppendMapItem(cfg, "filters", filters)
```

Comment:
(suggestion) for explicitness, I'd prefer to avoid relying on the CRD struct == the Prometheus struct. We also need to ensure that the generation is deterministic (see stringMapToMapSlice for example).

Reply:
I originally had the logic be explicit, but then changed it. I will change it back to how it was.